openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx

```diff
@@ -249,6 +291,203 @@ const GlossaryTermTab = ({
     return data;
   }, [glossaryTerms, permissions]);
 
+  const listOfVisibleColumns = useMemo(() => {
+    return ['name', 'description', 'owners', 'status', 'new-term'];
+  }, [permissions]);
+
+  const defaultCheckedList = useMemo(() => {
+    return columns
+      .filter((column) => listOfVisibleColumns.includes(column.key as string))
+      .map((column) => column.key)
+      .filter((key): key is string => key !== undefined);
+  }, [columns, listOfVisibleColumns]);
+
+  const [checkedList, setCheckedList] = useState<string[]>(defaultCheckedList);
+  const [tempCheckedList, setTempCheckedList] = useState<string[]>([
+    ...checkedList,
+  ]);
```

Comment:
@shrushti2000 , let's use some meaningful names , moreover why we need two states here `checkedList` and `tempCheckedList`

Reply:
done, have changed the variable names
checkedList-> selectedColumns
tempCheckedList -> columnDropdownSelections
Had created two separate states, 1 for maintaining the track of selections within the dropdown and 1 for keeping the track of final selection list after user clicks on save.